packages/website/docs/sources.md

```diff
@@ -180,15 +167,50 @@ const autocomplete = createAutocomplete({
 });
 ```
 
-You can notice that we can **get the `query` from the `getSources` function to conditionally return sources**.
+The [`getSources`](#getsources) function provides access to the current `query`, which you can use to return sources conditionally. You can use this pattern to display recent searches when the query is empty and search results when the user types.
 
-This pattern can be extended to display recent searches on empty query, and search results when a query is typed. You can compute dynamic sources based on the complete autocomplete state, not only the query.
+Note that you have access to the [complete autocomplete state](state), not only the query. It lets you compute sources based on many different aspects.
 
 ### Using asynchronous sources
 
-`getSources` also supports promises, which means that you can fetch your sources from an asynchronous API:
+The [`getSources`](#getsources) function supports promises so that you can fetch sources from any asynchronous API. It can be Algolia or any third-party API you can query with an HTTP request.
+
+For example, you could use the [Query Autocomplete](https://developers.google.com/places/web-service/query) service of the Google Places API to search for places and retrieve popular queries that map to actual points of interest.
+
+```js
+import { createAutocomplete } from '@algolia/autocomplete-core';
+
+const autocomplete = createAutocomplete({
+  getSources({ query }) {
+    return fetch(
```

Comment:
I've added an example using a third-party to showcase that Autocomplete isn't Algolia-exclusive.

The Google Places API and its Query Autocomplete service are an interesting example as they return both better queries and actual places, meaning we get a rather terse example where we can still showcase usage of `getItemInputValue`.

Reply:
Great example—did you try it with an actual API key?

Reply:
No because you have to provide billing info to put your hands on a key. Do you have one yourself? Otherwise I can ask around.

Reply:
You can try that one:

https://github.com/algolia/doc-code-samples/blob/ec699c32f572d946a705264e91243fa58229af4e/React%20InstantSearch/places/src/App.js#L38

Reply:
Already tried but it has a referrer restriction. It even doesn't work in Postman (bad request, not much help in the error message) with the example endpoint of their docs. My guess is that this API key doesn't have the  Places API enabled.

Reply:
Ah true, I remember now that we restricted referrers for that API key.

Reply:
Let's grab a key later on when we want to build demos—for this one we can build a demo and take a video of it so that we don't expose our API key and get billed for tons of requests.

The code snippet should be fine, and worst case scenario, someone will PR with a fix.